src/plugins/workflow_designer/src/library/DocWorkers.cpp

```diff
@@ -873,6 +875,143 @@ bool MSAWriter::isStreamingSupport() const {
+                    annotationName = QString("unknown features %1").arg(featuresNum);
+                }
+                att = qobject_cast<AnnotationTableObject *>(doc->findGObjectByName(annotationName));
+                if (att == nullptr) {
+                    doc->addObject(att = new AnnotationTableObject(annotationName, context->getDataStorage()->getDbiRef()));
+                    if (dna) {
+                        att->addObjectRelation(dna, ObjectRole_Sequence);
+                    }
+                }
+                algoLog.trace(QString("Adding features [%1] to UgeneDB doc %2").arg(annotationName).arg(doc->getURLString()));
+            }
+            att->addAnnotations(atl);
+        }
+    }
+}
+
+bool UgeneDBWriter::hasDataToWrite(const QVariantMap &data) const {
+    return SeqWriter::hasSequenceOrAnns(data);
+}
+
+QSet<GObject *> UgeneDBWriter::getObjectsToWrite(const QVariantMap &data) const {
+    return QSet<GObject *>() << SeqWriter::getSeqObject(data, context) << SeqWriter::getAnnObject(data, context);
+}
+
+void UgeneDBWriter::streamingStoreEntry(DocumentFormat *format, IOAdapter *io, const QVariantMap &data, WorkflowContext *context, int entryNum) {
+    U2OpStatus2Log os;
+    QScopedPointer<U2SequenceObject> seqObj(nullptr);
+    QString annotationName;
+    if (data.contains(BaseSlots::DNA_SEQUENCE_SLOT().getId())) {
+        seqObj.reset(getCopiedSequenceObject(data, context, os));
+        SAFE_POINT_OP(os, );
+
+        if (seqObj->getGObjectName().isEmpty()) {
+            seqObj->setGObjectName(QString("unknown sequence %1").arg(entryNum));
+            annotationName = QString("unknown features %1").arg(entryNum);
+        } else {
+            annotationName = getAnnotationName(seqObj->getGObjectName());
+        }
+    }
+
+    QList<GObject *> anObjList;
+    if (data.contains(BaseSlots::ANNOTATION_TABLE_SLOT().getId())) {
+        const QVariant &annsVar = data[BaseSlots::ANNOTATION_TABLE_SLOT().getId()];
+        const QList<SharedAnnotationData> atl = StorageUtils::getAnnotationTable(context->getDataStorage(), annsVar);
+
+        if (!atl.isEmpty()) {
+            if (annotationName.isEmpty()) {
+                annotationName = QString("unknown features %1").arg(entryNum);
+            }
+            AnnotationTableObject *att = new AnnotationTableObject(annotationName, context->getDataStorage()->getDbiRef());
+            anObjList << att;
+            att->addAnnotations(atl);
+        }
+    }
+
+    QMap<GObjectType, QList<GObject *>> objectsMap;
+    {
+        if (nullptr != seqObj.data()) {
+            QList<GObject *> seqs;
+            seqs << seqObj.data();
+            objectsMap[GObjectTypes::SEQUENCE] = seqs;
+        }
+        if (!anObjList.isEmpty()) {
+            objectsMap[GObjectTypes::ANNOTATION_TABLE] = anObjList;
+        }
+    }
+    CHECK(!objectsMap.isEmpty(), );
+
+    format->storeEntry(io, objectsMap, os);
+
+    foreach (GObject *o, anObjList) {
```

Comment:
On line 1003 you export this list content and now you are deleting it. Looks like a bug?

Reply:
It is just memory cleaning after creation objects at line 982. I think it should be replaced with qDeleteAll().